trame_server/core.py

```diff
@@ -559,7 +559,11 @@ def start(
         if exec_mode == "main":
             self._running_stage = 0
             if self.controller.on_server_exited.exists():
-                self.controller.on_server_exited(**self.state.to_dict())
+                loop = asyncio.get_event_loop()
+                for exit_task in self.controller.on_server_exited(
+                    **self.state.to_dict()
+                ):
+                    loop.run_until_complete(exit_task)
```

Comment:
That is not correct... It works for you because you only have tasks but try adding other functions that return something and/or nothing... You should check each `exit_task` to see if they are awaitable and if-so, then call `loop.run_until_complete`

Reply:
Right, those don't execute. Addressed the issue.